### ISSUE TITLE
[Enhancement] Let users rename datasets

### ISSUE THREAD

**Issue by rjimenezda** [keplergl/kepler.gl#792]
**Is your feature request related to a problem? Please describe.**
When importing data to kepler, the dataset label is fixed and cannot be changed. 

There are several cases as how this happens:

- When uploading a *file*, the filename will be used as a label.
- When loading demo data, it's always 'new dataset'.
- When loading via a URL, if it's a URI that points to a file, it's the filename, but when using something like an API, it's the *whole URL*[1].

**Describe the solution you'd like**
I would like to be able to label them myself, much like we can label layers. File names might be too similar, not 'nice' looking, or too long for certain views (like the table view)

**Describe alternatives you've considered**
I can think of two approaches:

1. Using an inline editor, like the layer's.

2. Using an inline action, like show table data / delete. This could lead to a small modal, or triggering an inline editor after clicking it.

The problem with 1. is that the component is used on three separate places, so it should be decided whether renaming can only be done everywhere, or just on the Layers screen. This adds complexity to the component which is quite straightforward right now.

[1] This worked up until #780, I've documented this issue on #791

**Comment by bkowshik** [comment]
Yes. This a super handy feature, particularly to make more human-friendly maps giving good names to the data layers. In this example, I have two datasets one from a Presto query that has named the file on the timestamp when the query was made and then a temporary file on my system.

<img width="1678" alt="Screenshot 2020-09-05 at 2 13 41 PM" src="https://user-images.githubusercontent.com/2899501/92301519-255f2200-ef82-11ea-86e0-5339b1426784.png">
